File: Assets/Scripts/Volume.cs

// Hannes Helmholz
//
// 

using UnityEngine;

[RequireComponent (typeof(Collider))]

/** Klasse, welche sich um die Lautstärke der
 *  AudioClips in der Anwendung kümmert.
 *  Es wird der Name des AudioClips und die
 *  Entfernung zum Player ermittelt, um einen
 *  3D-Sound auszugeben.
 */
public class Volume : MonoBehaviour
{
 
    // =============================================================================
    // MEMBERS ---------------------------------------------------------------------
    [SerializeField]
    private string Name;
    private Collider Coll;
    // =============================================================================
 
 
 
    // =============================================================================
    // METHODS UNITY ---------------------------------------------------------------
 
    void Awake ()
    {
        Coll = collider;
    }
    // =============================================================================
 
 
 
    // =============================================================================
    // METHODS ---------------------------------------------------------------------
 
    public VolumeData GetVolumeData ( Vector3 point )
    {
        float distance = Coll.bounds.SqrDistance ( point );
        return new VolumeData ( Name, distance );
    }
    // =============================================================================
}

public class VolumeData
{
    public string Name;
    public float Distance;

 
    public VolumeData ( string name, float distance )
    {
        this.Name = name;
        this.Distance = distance;
    }
}

File: ISS Emergency/Assets/Scripts/Volume.js
#pragma strict

// Hannes Helmholz
//
// 


@script RequireComponent(Collider)

private class Volume extends MonoBehaviour {
	
	// =============================================================================
	// MEMBERS ---------------------------------------------------------------------
	@SerializeField private var Name:String;
	
	private var Coll:Collider;
	// =============================================================================
	
	
	
	// =============================================================================
	// METHODS UNITY ---------------------------------------------------------------
	
	function Awake() {
		Coll = collider;
	}
	// =============================================================================
	
	
	
	// =============================================================================
	// METHODS ---------------------------------------------------------------------
	
	public function GetVolumeData(point:Vector3):VolumeData {
		var distance:float = Coll.bounds.SqrDistance(point);
		return VolumeData(Name, distance);
	}
	// =============================================================================
}



private class VolumeData extends System.ValueType {
	public var Name:String;
	public var Distance:float;
	
	public function VolumeData(name:String, distance:float) {
		this.Name = name;
		this.Distance = distance;
	}
}